### DISCUSSION TITLE
Gap Between Start and End of Curved Text in Circular Path Rendering in Fabric.js

### DISCUSSION THREAD

**Discussion by Arkit-Sutariya** [fabricjs/fabric.js#10237] [Q&A]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### Node Version (if applicable)

16.14.0

### Link To Reproduction

No

### Steps To Reproduce

I'm working with Fabric.js curved text, and I've provided the code in this [sandbox](https://codesandbox.io/p/sandbox/sharp-https-xcvqd8). In the current implementation, I'm converting standard text into curved text using additional logic, but the output isn't as expected. Could you provide the best solution to achieve the desired result?

### Expected Behavior

My expected output is :: 
![image](https://github.com/user-attachments/assets/bc9c749a-8dc9-4dcb-8490-dedbde42692a)

### Actual Behavior

Actual output is :: 
![image](https://github.com/user-attachments/assets/1739c235-428d-4ebc-b04a-2191ed6752f7)

**Comment by mhieupham1** [comment]
![Screenshot 2024-10-23 at 09 23 17](https://github.com/user-attachments/assets/e807d8d8-50b5-4170-9328-e2dcb7053308)
change fontSize to get it

**Reply by Arkit-Sutariya** [reply]
Thank you for your response. I appreciate your assistance, but the provided solution doesn't fully address the issue at hand. Allow me to offer more context for clarity.

We are working on a tool that allows users to design various types of flyers and download them in either image or PDF format. One of the features in this tool is curved text, which displays correctly during the design phase.

However, when I use the JSON data from the user's design to generate a PDF (using a separate library), there is a noticeable gap between the start and end of the curved text when I load the design with the canvas.loadFromJSON() method. This issue arises without any changes to the font size.

For further clarity, I have included the complete JSON data of the curved text in the objectrender variable. I would appreciate any guidance on how to resolve this issue while maintaining the current font size settings.

**Reply by asturur** [reply]
you have to include the full app including the pdf export process.
How do you think people are going to figure out what is wrong.
Do you also understand that in the opening post you omitted that PDF and export/import was involved?

**Reply by Arkit-Sutariya** [reply]
I'm working on a Node and Angular project. When a user selects to export a PDF or Image, I call an API on the Node server, passing JSON data as provide in curveTextData variable.

Once the data is received on the Node server, I load an HTML file (which I’ve shared with you) that calls a JavaScript file (also provided).

Due to restrictions, I’m unable to share the full working application at this time, as development is currently on a local server.
